### DISCUSSION TITLE
Feedback loop for chat models

### DISCUSSION THREAD

**Discussion by saifu19** [FlowiseAI/Flowise#1024] [Q&A] [answered]
Is there some kind of feedback loop which I can use in my flowise model ? I have implemented a model with my pinecone retreiver, memory buffer, openai chat model and chat retreiver agent. I want to introduce a feedback loop so that when the chatbot produces an incorrect response, I correct it and then the chatbot must learn it and respond to all future relevant prompts with the correct data. Memory buffer will store it but only for the current chat session and I want something permanent here.
If this is not possible, then is there any way to store chat in a file so that I can later clean the data and use it to fine tune my models ?

**Comment by HenryHengZJ** [comment]
you can use 3rd party memory node like Zep, DynamoDB, Redis to store the chat messages - https://docs.flowiseai.com/memory/zep-memory

**Reply by saifu19** [reply]
That will only serve as memory. I want to collect the data of user inputs and chatbot outputs for future use. I have connected a dynamodb table and it works perfectly as memory but I don't see any entries in my dynamodb table.

**Reply by HenryHengZJ** [reply, marked as answer]
there was a bug, but it was fixed - https://github.com/FlowiseAI/Flowise/pull/1025 , can you update to latest version?
